compiler/rustc_codegen_ssa/src/back/archive.rs

```diff
@@ -307,10 +307,15 @@ impl<'a> ArchiveBuilder for ArArchiveBuilder<'a> {
             let file_name = String::from_utf8(entry.name().to_vec())
                 .map_err(|err| io::Error::new(io::ErrorKind::InvalidData, err))?;
             if !skip(&file_name) {
-                self.entries.push((
-                    file_name.into_bytes(),
-                    ArchiveEntry::FromArchive { archive_index, file_range: entry.file_range() },
-                ));
+                if entry.is_thin() {
+                    let member_path = archive_path.parent().unwrap().join(Path::new(&file_name));
```

Comment:
Question: can this ever unwrap and result in an ICE?

Reply:
`parent()` can only return `None` when the given path is `/` or an empty string, neither of which should allow the `File::open(&archive_path)` above to succeed. It's technically possible if you intercept syscalls or replace libc, but I don't think we should defend against that.

Reply:
Thanks for clarifying ^^